src/experiments.js

```diff
@@ -76,6 +76,20 @@ export function isExperimentOn(win, experimentId) {
   return !!toggles[experimentId];
 }
 
+/**
+ * Assert that either the global or component Bento experiment is on.
+ * @param win
+ * @param {string} componentName
+ * @throws {UserError} if neither experiment is enabled
+ */
+export function assertBentoExperiment(win, componentName) {
+  userAssert(
+    isExperimentOn(win, 'bento') ||
+      isExperimentOn(win, /*OK*/ `bento-${componentName}`),
```

Comment:
Are there compilation steps that depend on this bit being static?

Reply:
At least the [experiment sweep tool does](https://github.com/ampproject/amphtml/blob/master/build-system/tasks/sweep-experiments/), but the restriction predates it.

Reply:
Yes, we've had issues where experiment code was missed because the name wasn't a literal string.

Reply:
Does that mean this PR is largely impossible? Would it work if the `bento-fit-text` etc. was a string provided by the component, or does it have to be a string in this call?

Reply:
This specific part, yes. It must be literal source text `isExperiemntOn(anything, 'anything')`. If there's not a literal string in the second arg (meaning it has `''` or `""` building a `StringLiteral` AST node), it's an error.

Reply:
I know it's a lint error if that's the case based on `tasks/is-experiment-on.js`, but what I'm wondering is what happens if it's not? The value passed is a string template literal, and I can't find any transforms in `build-system` or elsewhere that seem to look for `isExperimentOn` specifically or require this to be static. 

Would this not be safe to do, if it's needed for every bento component? It doesn't seem like something the experiment sweep tool would need to be able to detect

Reply:
Shuffled things around so `PreactBaseElement` does most of the work, while each AMP element can define their own `isComponentExperimentOn`, lmk if you think it's worthwhile